src/API/arkode.jl

```diff
@@ -66,16 +66,16 @@ function ARKStepResFtolerance(arkode_mem, rfun::ARKRwtFn)
     ccall((:ARKStepResFtolerance, libsundials_arkode), Cint, (ARKStepMemPtr, ARKRwtFn), arkode_mem, rfun)
 end
 
-function ARKStepSetLinearSolver(arkode_mem, LS::SUNLinearSolver, A::SUNMatrix)
+function ARKStepSetLinearSolver(arkode_mem, LS::SUNLinearSolver, A)
```

Comment:
it's always dangerous directly modifying the wrapped API files. Instead, make an API additions file for these dispatches.

Reply:
We should probably do the same here as with NVector and N_Vector probably and have a conversion method and add another function that dispatches on the type.

Reply:
We can add some conversion method for these function and have a similar behavior as N_Vector and NVector also.